web/ui/react-app/src/utils/index.ts

```diff
@@ -197,7 +197,7 @@ export const parseOption = (param: string): Partial<PanelOptions> => {
       return { type: decodedValue === '0' ? PanelType.Graph : PanelType.Table };
 
     case 'stacked':
-      return { stacked: decodedValue === '1' };
+      return { displayMode: decodedValue === '1' ? GraphDisplayMode.Stacked : GraphDisplayMode.Lines };
```

Comment:
We will always want to encode all available graph options in the URL so that we can send around heatmap graphs as well, so we'll want to encode the new display mode setting as well.

I'd go for this approach:

* On reading, be backwards-compatible with the legacy `stacked` parameter and just translate it to the new display mode setting as you are doing it here already.
* On writing the options back to the URL, only store the display mode parameter (maybe `display_mode=lines|stacked|heatmap`?) and omit `stacked`.

If a URL happens to somehow contain both the legacy `stacked` parameter and `display_mode`, then `display_mode` can take precedence.

That way a new Prometheus version can still open the old URLs correctly, but will produce the new format going forward.

Reply:
What do you think, what's the better practice for test cases: using string literals or enums for parameters like 'display_mode'?

```
.toEqual(...g0.display_mode=${GraphDisplayMode.Stacked}&...);
or
.toEqual(...g0.display_mode=stacked&...);"
```

Reply:
Yeah, I get the question for tests, but I don't think it makes much of a difference in this kind of case, so I'm happy with either.